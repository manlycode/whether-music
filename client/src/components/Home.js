import React, { Component } from 'react'
import Form from './Form'
import axios from 'axios';
import MusicBoxContainer from './MusicBoxContainer';

export default class Home extends Component {
  state = {
    city: '',
    weatherData: [],
    musicData: []
  }

  handleInputChange = event => {
    const{ name, value } = event.target;
    this.setState({
      [name]: value
    })
    this.handleFormSubmit = this.handleFormSubmit.bind(this)
    this.handleInputChange = this.handleInputChange.bind(this)
  }

  handleFormSubmit = event => {
    event.preventDefault()
    axios
    .get('/api/weather', { params: { city: this.state.city } })
    .then(results => {
      this.setState({
        weatherData: results.data
      })
      return axios.post('/api/songs', { params: { weatherData: this.state.weatherData } })
    })
    .then(results => {
      this.setState({
        musicData: results.data
      })
      console.log(results.data);
      
    })
    .catch(error => {
      console.log(error);
    })
  }

  render() {
    const { city, musicData } = this.state;
    return (
      <div className='container'>
        <Form
          inputChange={this.handleInputChange}
          formSubmit={this.handleFormSubmit}
          city={city}
        />
        {
          musicData !== null && !musicData.length ? (
            <h2>No results to display</h2>
          ) : (
              <MusicBoxContainer
                musicData={musicData}
              />
            )
        }
      </div>
    )
  }
}
