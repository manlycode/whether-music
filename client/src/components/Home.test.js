import React from 'react'
import { mount, shallow } from 'enzyme'
import Home from './Home'
import { isThisHour } from 'date-fns';


describe('Home', () => {
  let wrapper
  beforeEach(() => wrapper = shallow(<Home />))

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should render a div', () => {
    expect(wrapper.find('div').length).toEqual(3)
  })
})