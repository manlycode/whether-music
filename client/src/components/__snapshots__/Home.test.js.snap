// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form>
        <div
          className="form-group"
        >
          <label
            htmlFor="Weather"
          >
            <strong>
              Weather
            </strong>
          </label>
          <input
            className="form-control"
            id="City"
            name="city"
            onChange={[Function]}
            placeholder="Enter a city"
            required={true}
            type="text"
            value=""
          />
        </div>
        <div
          className="pull-right"
        >
          <button
            className="btn btn-lg btn-danger float-right"
            onClick={[Function]}
            type="submit"
          >
            Search
          </button>
        </div>
      </form>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="form-group"
          >
            <label
              htmlFor="Weather"
            >
              <strong>
                Weather
              </strong>
            </label>
            <input
              className="form-control"
              id="City"
              name="city"
              onChange={[Function]}
              placeholder="Enter a city"
              required={true}
              type="text"
              value=""
            />
          </div>,
          <div
            className="pull-right"
          >
            <button
              className="btn btn-lg btn-danger float-right"
              onClick={[Function]}
              type="submit"
            >
              Search
            </button>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label
                htmlFor="Weather"
              >
                <strong>
                  Weather
                </strong>
              </label>,
              <input
                className="form-control"
                id="City"
                name="city"
                onChange={[Function]}
                placeholder="Enter a city"
                required={true}
                type="text"
                value=""
              />,
            ],
            "className": "form-group",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <strong>
                  Weather
                </strong>,
                "htmlFor": "Weather",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Weather",
                },
                "ref": null,
                "rendered": "Weather",
                "type": "strong",
              },
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "form-control",
                "id": "City",
                "name": "city",
                "onChange": [Function],
                "placeholder": "Enter a city",
                "required": true,
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <button
              className="btn btn-lg btn-danger float-right"
              onClick={[Function]}
              type="submit"
            >
              Search
            </button>,
            "className": "pull-right",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Search",
              "className": "btn btn-lg btn-danger float-right",
              "onClick": [Function],
              "type": "submit",
            },
            "ref": null,
            "rendered": "Search",
            "type": "button",
          },
          "type": "div",
        },
      ],
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form>
          <div
            className="form-group"
          >
            <label
              htmlFor="Weather"
            >
              <strong>
                Weather
              </strong>
            </label>
            <input
              className="form-control"
              id="City"
              name="city"
              onChange={[Function]}
              placeholder="Enter a city"
              required={true}
              type="text"
              value=""
            />
          </div>
          <div
            className="pull-right"
          >
            <button
              className="btn btn-lg btn-danger float-right"
              onClick={[Function]}
              type="submit"
            >
              Search
            </button>
          </div>
        </form>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="form-group"
            >
              <label
                htmlFor="Weather"
              >
                <strong>
                  Weather
                </strong>
              </label>
              <input
                className="form-control"
                id="City"
                name="city"
                onChange={[Function]}
                placeholder="Enter a city"
                required={true}
                type="text"
                value=""
              />
            </div>,
            <div
              className="pull-right"
            >
              <button
                className="btn btn-lg btn-danger float-right"
                onClick={[Function]}
                type="submit"
              >
                Search
              </button>
            </div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <label
                  htmlFor="Weather"
                >
                  <strong>
                    Weather
                  </strong>
                </label>,
                <input
                  className="form-control"
                  id="City"
                  name="city"
                  onChange={[Function]}
                  placeholder="Enter a city"
                  required={true}
                  type="text"
                  value=""
                />,
              ],
              "className": "form-group",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <strong>
                    Weather
                  </strong>,
                  "htmlFor": "Weather",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Weather",
                  },
                  "ref": null,
                  "rendered": "Weather",
                  "type": "strong",
                },
                "type": "label",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "form-control",
                  "id": "City",
                  "name": "city",
                  "onChange": [Function],
                  "placeholder": "Enter a city",
                  "required": true,
                  "type": "text",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <button
                className="btn btn-lg btn-danger float-right"
                onClick={[Function]}
                type="submit"
              >
                Search
              </button>,
              "className": "pull-right",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Search",
                "className": "btn btn-lg btn-danger float-right",
                "onClick": [Function],
                "type": "submit",
              },
              "ref": null,
              "rendered": "Search",
              "type": "button",
            },
            "type": "div",
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
